File: leetcode/AIcoding/5.Qwen-3/erlang_solutions/2453-destroy-sequential-targets.erl
-module(solution).
-export([destroy_targets/2]).

destroy_targets(Targets, Cost) ->
    destroy_targets(Targets, Cost, 0, 0).

destroy_targets([], _, _, Result) ->
    Result;
destroy_targets([H | T], Cost, Index, Result) ->
    case H of
        {Start, End} when Start =< Index andalso End >= Index ->
            destroy_targets(T, Cost, Index + 1, Result + Cost);
        _ ->
            destroy_targets(T, Cost, Index + 1, Result)
    end.

File: leetcode/AIcoding/5.Qwen-3/cpp_solutions/2453-destroy-sequential-targets.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <unordered_map>

using namespace std;

class Solution {
public:
    int findRightmostTarget(vector<int>& targets, int start) {
        int n = targets.size();
        for (int i = n - 1; i >= 0; --i) {
            if (targets[i] == start) {
                return i;
            }
        }
        return -1;
    }

    int destroyTargets(vector<int>& targets, int space) {
        int n = targets.size();
        unordered_map<int, int> countMap;
        for (int i = 0; i < n; ++i) {
            int remainder = targets[i] % space;
            countMap[remainder]++;
        }

        int maxCount = 0;
        for (auto& [key, value] : countMap) {
            if (value > maxCount) {
                maxCount = value;
            }
        }

        int result = -1;
        for (int i = 0; i < n; ++i) {
            int remainder = targets[i] % space;
            if (countMap[remainder] == maxCount) {
                int rightmost = findRightmostTarget(targets, targets[i]);
                if (rightmost != -1) {
                    result = targets[rightmost];
                    break;
                }
            }
        }

        return result;
    }
};